base/Dockerfile

```diff
@@ -2,11 +2,34 @@
 FROM gcr.io/google-appengine/debian8
 
 # Install updates and dependencies
-RUN apt-get update -y && apt-get install --no-install-recommends -y -q curl python build-essential git ca-certificates libkrb5-dev imagemagick netbase && \
-    apt-get clean && rm /var/lib/apt/lists/*_*
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y -q \
+      build-essential \
+      ca-certificates \
+      curl \
+      git \
+      imagemagick \
+      libkrb5-dev \
+      netbase \
+      python && \
+    apt-get clean && \
+    rm /var/lib/apt/lists/*_*
+
+# Add the files necessary for verifying and installing node
+# These files are placed in /opt/google_appengine/runtime so that
+# they will remain even if the /nodejs directory is removed.
+# Further, the PATH is is not updated, because these scripts should
+# only be called using an absolute path.
+ADD contents/ /opt/google_appengine/runtime/
+RUN ln -s /opt/google_appengine/runtime/install_node /usr/local/bin/install_node
 
 # Install the latest LTS release of nodejs
-RUN mkdir /nodejs && curl https://nodejs.org/dist/v6.10.3/node-v6.10.3-linux-x64.tar.gz | tar xvzf - -C /nodejs --strip-components=1
+RUN /opt/google_appengine/runtime/verify_install_node \
+    https://nodejs.org/dist/v6.10.3 \
+    v6.10.3 \
+    SHASUMS256.txt.asc \
+    /nodejs \
+    FAIL_ON_ERROR
```

Comment:
nit: positional arguments are not very readable. Most of these could be derived from the version being installed anyway.